min-image/runtimes/python/ol.c

```diff
@@ -8,6 +9,71 @@
 #include <sys/types.h>
 #include <sys/wait.h>
 #include <seccomp.h>
+#include <sys/capability.h>
+
+static int modify_cap_impl(int cap, int setting) {
+  cap_t caps;
+  cap_value_t capList[1];
+  int rc = 0;
+
+  caps = cap_get_proc();
+  if (caps == NULL) {
+    rc = -1;
+    goto out;
+  }
+
+  capList[0] = cap;
+  if (cap_set_flag(caps, CAP_EFFECTIVE, 1, capList, setting) == -1) {
+    cap_free(caps);
```

Comment:
One of the nice things about a goto is it gives you a single place to clean things up, like this:

if (caps != NULL)
    cap_free(caps);

Reply:
@tylerharter I actually didn't what you mean here. I tried to use the goto statement already. Is there an even better way to do it?

Reply:
You have cap_free in multiples places.  What I mean is that cap_free can be after "out:".  You'll need to check whether it is NULL before freeing there, of course.